WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs

```diff
@@ -1321,357 +1327,332 @@ private async Task<JObject> HandleMergeBranchActionAsync(DataRow dataRowWithSett
-                                    }
+                                historyItemsSynchronised.Add(historyId);
+                                itemsProcessed++;
+                                await UpdateProgressInQueue(databaseConnection, queueId, itemsProcessed);
 
-                                    break;
-                                }
-                            }
-                            else
-                            {
-                                linkCacheData = new BranchMergeLinkCacheModel
-                                {
-                                    Id = linkId.Value
-                                };
-                                linksCache.Add(linkCacheData);
+                                actionData.Status = ObjectActionMergeStatuses.SkippedAndRemoved;
+                                actionData.MessageBuilder.AppendLine("The current row was skipped and removed, because the link has been deleted, which means that there is no point in merging that.");
+                                branchBatchLoggerService.LogMergeAction(actionData);
+                                continue;
                             }
 
-                            await using var branchCommand = branchConnection.CreateCommand();
-                            AddParametersToCommand(sqlParameters, branchCommand);
-
-                            // Replace wiser_itemlinkdetail with wiser_itemlink because we need to get the source and destination from [prefix]wiser_itemlink, even if this is an update for [prefix]wiser_itemlinkdetail.
-                            branchCommand.CommandText = $"SELECT type, item_id, destination_item_id FROM `{tableName.ReplaceCaseInsensitive(WiserTableNames.WiserItemLinkDetail, WiserTableNames.WiserItemLink)}` WHERE id = ?linkId";
-                            var linkDataTable = new DataTable();
-                            using var branchAdapter = new MySqlDataAdapter(branchCommand);
-                            branchAdapter.Fill(linkDataTable);
-                            if (linkDataTable.Rows.Count == 0)
+                            if (linkCacheData.Id > 0)
                             {
-                                branchCommand.CommandText = $"SELECT type, item_id, destination_item_id FROM `{tableName.ReplaceCaseInsensitive(WiserTableNames.WiserItemLinkDetail, WiserTableNames.WiserItemLink)}{WiserTableNames.ArchiveSuffix}` WHERE id = ?linkId";
-                                branchAdapter.Fill(linkDataTable);
-                                if (linkDataTable.Rows.Count == 0)
+                                actionData.ItemIdOriginal = linkCacheData.ItemId ?? 0;
+                                actionData.ItemIdMapped = linkCacheData.ItemId ?? 0;
+                                actionData.LinkDestinationItemIdOriginal = linkCacheData.DestinationItemId ?? 0;
+                                actionData.LinkDestinationItemIdMapped = linkCacheData.DestinationItemId ?? 0;
```

Comment:
Is it correct both properties use the same value?

Reply:
Yes that is correct; We want to make sure that we always have a mapped ID. So we always fill the original ID there first, then later we check if that has a mapped ID and if yes, we change it. If it doesn't have a mapped ID, then it will stay the same as the original ID. This way we can just use the mapped ID whenever we need to use that ID in the production database and we're sure it always has a value. See lines 1720+ in this class where we overwrite these values with the actual mapped IDs, if they exist.